doc/adr/0011-concept-exception-handling-backend.md

```diff
@@ -0,0 +1,33 @@
+# 10. How to use exceptions in the backend
+
+Date: 2024-06-28
+
+## Status
+
+Accepted
+
+## Context
+So far we used different approaches when to use Optionals, throw checked exceptions or unchecked exceptions. This also led
+to using exceptions from other layers. We want to streamline our approach.
+
+## Decision
+Layer-Specific Exceptions
+* By having each layer (ports, adapters, and application) define its own exceptions, we achieve a clear separation of concerns. This ensures that exceptions are relevant to the specific context in which they occur.
+* Avoiding a common exception package prevents mixing unrelated exceptions and promotes a more organized codebase.
+
+Custom Runtime Exceptions
+* Throwing custom runtime exceptions in the application layer allows us to encapsulate domain-specific logic. These exceptions can be tailored to the norms project’s requirements.
+* Using @ControllerAdvice to handle these exceptions ensures consistent error handling across the application. It centralizes exception handling logic and provides a single point of control.
+
+Output Ports
+* Returning optionals from output ports aligns with functional programming principles. Optionals allow for more elegant handling of absence or optional results.
+* Translating exceptions thrown in output ports at the application layer maintains a clean boundary between layers. It avoids leaking implementation details and promotes a clear contract.
```

Comment:
I would maybe add here, since we don't have any exceptions being thrown in the output ports, that in case we want to throw an exception we would translate it from the output port to an application exception